frontend/src/components/misc/UserPreferences.tsx

```diff
@@ -156,86 +158,101 @@ class JsonViewerTab extends Component {
-                </Button>
-            </Label>
 
-            <Label text="Reset">
-                <>
-                    <div>
-                        <Input style={{ maxWidth: '360px', marginRight: '8px', fontFamily: 'monospace', fontSize: '0.85em' }} spellCheck={false}
-                            placeholder='type "reset" here to confirm and enable the button'
-                            value={this.resetConfirm}
-                            onChange={str => this.resetConfirm = str.target.value} />
-                        <Button onClick={() => {
-                            clearSettings();
-                            message.success('All settings have been reset to their defaults');
-                            this.resetConfirm = '';
-                        }} colorScheme="red" disabled={this.resetConfirm != 'reset'}>Reset</Button>
-                    </div>
-                    <span className="smallText">Clear all your user settings, resetting them to the default values</span>
-                </>
-            </Label>
-        </>;
-    }
-}
+                }}>Import</Button>
+            </Flex>
+        </Label>
+
+        <Label text="Export">
+            <Button onClick={() => {
+                navigator.clipboard.writeText(JSON.stringify(uiSettings)).then(() => {
+                    toast({
+                        status: 'success',
+                        description: 'Preferences copied to clipboard!'
+                    })
+                }).catch((e) => {
+                    toast({
+                        status: 'error',
+                        description: 'Unable to copy settings to clipboard. See console for more information.'
+                    });
+                    console.error('unable to copy settings to clipboard', {error: e});
+                })
+            }}>
+                Export User Preferences
+            </Button>
+        </Label>
+
+        <Label text="Reset">
+            <>
+                <div>
+                    <Input style={{ maxWidth: '360px', marginRight: '8px', fontFamily: 'monospace', fontSize: '0.85em' }} spellCheck={false}
+                           placeholder='type "reset" here to confirm and enable the button'
+                           value={$state.resetConfirm}
+                           onChange={str => $state.resetConfirm = str.target.value} />
+                    <Button onClick={() => {
+                        clearSettings();
+                        toast({
+                            status: 'success',
+                            description: 'All settings have been reset to their defaults'
+                        });
+                        $state.resetConfirm = '';
+                    }} colorScheme="red" isDisabled={$state.resetConfirm !== 'reset'}>Reset</Button>
+                </div>
+                <span className="smallText">Clear all your user settings, resetting them to the default values</span>
```

Comment:
[suggestion] [future] After we start using 3.24.0, maybe we can start using the Text component more that will eventually have variants, provided by @frankieshakes